Move generate graph function out of student report pdf js file

gabivoicu opened the pull request:
# Who is this PR for?
Developers primarily for code quality, users secondarily for more stability.

# What problem does this PR fix?
Lack of test for graph generator function in student report pdf.

# What does this PR do?
Moves graph generator function into a separate file, refactors that file a little bit, adds smoke test for function.

# How to test this PR?
Check that the charts in the student report PDF still display the same values in the same manner as before this PR.

# Checklists

## Javascript QA

*If the PR touches Javascript, follow this checklist. If the PR doesn't touch Javascript, you can delete this section.*

*Which features or pages does this PR touch? Make a list here so that you can double-check each affected page in a Internet Explorer before merging.*

*Make one list for the author and another for the reviewer.*

*For more info on how to download and use a Virtual Machine to test in IE, see the README.*

*Example:*

+ [x] Author checked latest in IE - Student Report PDF
+ [x] Reviewer checked latest in IE - Student Report PDF

## Automated Testing

*Does the PR touch an area of the codebase that needs more specs? More examples within the specs? More scenarios covered by specs?*

+ [x] Author improved specs for code in need of better test coverage

## Screenshots
### IE 11 Windows 7 Absences & Tardies Chart
<img width="650" alt="ie11-win7-at" src="https://user-images.githubusercontent.com/8533377/35170169-dbe90606-fd24-11e7-832d-fd68eaad6698.png">

### IE 11 Windows 7 Discipline Incidents Chart
<img width="638" alt="ie11-win7-di" src="https://user-images.githubusercontent.com/8533377/35170213-0423702a-fd25-11e7-9226-bd9dc0e66a24.png">

### IE 11 Windows 8 Absences & Tardies Chart
<img width="588" alt="ie11-win8-at" src="https://user-images.githubusercontent.com/8533377/35170224-0b6f171c-fd25-11e7-80f1-6aa886f91ded.png">

### IE 11 Windows 8 Discipline Incidents Chart
<img width="591" alt="ie11-win8-di" src="https://user-images.githubusercontent.com/8533377/35170236-15a39730-fd25-11e7-9bd7-08e6caf4236b.png">

alexsoble commented on app/assets/javascripts/student_profile/pdf/GenerateReportGraph.js:
```diff
@@ -0,0 +1,59 @@
+          color: (Highcharts.theme && Highcharts.theme.textColor) || 'gray'
+        },
+        formatter: function(){
+          const val = this.y < 1 ? '' : this.y;
```
@gabivoicu Can you double check this one? I think we want `this.total` here instead of `this.y`.

gabivoicu replied:
Can't see my response on the front page so adding a link to it [here](https://github.com/studentinsights/studentinsights/pull/1374/files#diff-9970c0bc40ac1427e00ff25192eb7832L36)

gabivoicu commented on app/assets/javascripts/student_profile/pdf/student_profile_pdf.js:
```diff
@@ -1,74 +1,10 @@
-            enabled: true,
-            color: (Highcharts.theme && Highcharts.theme.dataLabelsColor) || 'white',
-            formatter: function(){
-              const val = this.y;
```
@alexsoble I refactored it from here so I thought `this.y` is correct. Is there something I am missing? 🤔

alexsoble commented:
Hey there @gabivoicu! Can I ask you for one thing? Would you mind taking screenshots of that the Student Report PDF looks like on this branch and adding those images to the PR description up top? Would also be helpful to me as the reviewer if you could addd a little explanation about the two `formatter` functions you refactored, what each one does, and where we should look on the PDF screenshot to see what they're doing.

gabivoicu commented:
hey @alexsoble, I've added the screenshots and comments about the formatter, caught another issue with it, and also added a `How to test this PR?` section to the PR to explicitly tell reviewer what to look at.

alexsoble commented:
@gabivoicu Thank you! 😎

alexsoble commented:
@gabivoicu Ooooh yeah always got to remember to `return`, great catch there.

alexsoble commented:
@gabivoicu I QA'ed in an IE VM — looks good! Thanks for adding those code comments, those functions were kind of mysterious to me before but now they're clearer. 

🚢
